tracer/test/Datadog.Trace.IntegrationTests/FleetInstaller/FleetInstallerTestsBase.cs

```diff
@@ -0,0 +1,121 @@
+﻿// <copyright file="FleetInstallerTestsBase.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NETFRAMEWORK
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Net.Http;
+using System.Security.Principal;
+using System.Threading.Tasks;
+using Datadog.FleetInstaller;
+using Datadog.Trace.TestHelpers;
+using FluentAssertions;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.IntegrationTests.FleetInstaller;
+
+public abstract class FleetInstallerTestsBase : IDisposable
+{
+    protected const string PreviousTracerVersion = "3.11.0";
+    private static bool? _isElevated;
+    private readonly ITestOutputHelper _output;
+    private List<string> _homeDirectory;
+
+    protected FleetInstallerTestsBase(ITestOutputHelper output)
+    {
+        Log = new(output);
+        _output = output;
+    }
+
+    public static bool IsRunningAsAdministrator
+    {
+        get
+        {
+            _isElevated ??= new WindowsPrincipal(WindowsIdentity.GetCurrent()).IsInRole(WindowsBuiltInRole.Administrator);
+            return _isElevated.Value;
+        }
+    }
+
+    protected FleetInstallerLogger Log { get; }
+
+    public void Dispose()
+    {
+        if (_homeDirectory is { } list)
+        {
+            foreach (var dir in list)
+            {
+                try
+                {
+                    Directory.Delete(dir, recursive: true);
+                    GacInstaller.TryGacUninstall(Log, new TracerValues(dir));
+                }
+                catch
+                {
+                    // Ignore, something is locking it
+                }
+            }
+        }
+    }
+
+    protected string CreateMonitoringHomeCopy()
```

Comment:
The monitoring home is quite big, we should probably cleanup after running each test

Reply:
We do 🙂 Everytime we create a copy, we add it to the `_homeDirectory` collection, and in the `Dispose` we delete the directory